packages/components/src/textarea/src/components/osds-textarea/osds-textarea.tsx

```diff
@@ -1,289 +1,136 @@
+    this.internalError = await this.controller.hasError();
   }
 
-  /**
-   * @see OdsTextAreaBehavior.emitChange
-   */
-  emitChange(value: HTMLTextAreaElement['value'], oldValue?: HTMLTextAreaElement['value']): void {
+  async emitChange(value: string, oldValue?: string): Promise<void> {
     this.odsValueChange.emit({
-      oldValue: oldValue === null ? oldValue : `${oldValue}`,
-      validity: this.controller.getTextAreaValidity(),
-      value: value === null ? value : `${value}`,
+      name: this.name,
+      oldValue: oldValue,
+      validity: await this.getValidity(),
+      value,
     });
   }
 
-  /**
-   * @see OdsTextAreaBehavior.emitBlur
-   */
-  emitBlur(): void {
-    this.odsBlur.emit();
+  formResetCallback(): Promise<void> {
+    return this.reset();
   }
 
-  /**
-   * @see OdsTextAreaBehavior.emitFocus
-   */
-  emitFocus(): void {
-    this.odsFocus.emit();
-  }
-
-  /**
-   * @see OdsTextAreaBehavior.onBlur
-   */
   onBlur(): void {
-    this.controller.onBlur();
+    this.odsBlur.emit();
   }
 
-  /**
-   * @see OdsTextAreaBehavior.onInput
-   */
   onInput(event: Event): void {
     this.controller.onInput(event);
   }
 
-  /**
-   * @see OdsTextAreaBehavior.onChange
-   */
-  onChange(): void {
-    this.controller.onChange();
-  }
-
-  /**
-   * @see OdsTextAreaBehavior.onFocus
-   */
-   @Listen('focus')
-  onFocus(): void {
-    this.controller.onFocus();
-    this.setFocus.bind(this)();
-  }
-
-  /**
-   * @see OdsTextAreaMethods.setFocus
-   */
-  @Method()
-   async setFocus(): Promise<void> {
-     this.textInputEl?.focus();
-     this.hasFocus = true;
-   }
-
-  /**
-   * @see OdsTextAreaMethods.getValidity
-   */
-  @Method()
-  async getValidity(): Promise<OdsTextAreaValidityState> {
-    return this.controller.getTextAreaValidity();
-  }
-
-  /**
-   * @see OdsTextAreaMethods.clear
-   */
-  @Method()
-  async clear(): Promise<void> {
-    this.controller.setValue();
-  }
-
-  /**
-   * @see OdsTextAreaMethods.reset
-   */
-  @Method()
-  async reset(): Promise<void> {
-    this.controller.setValue(this.defaultValue || '');
-  }
-
-  /**
-   * @see OdsTextAreaMethods.setTextAreaTabindex
-   */
-  @Method()
-  async setTextAreaTabindex(value: number): Promise<void> {
-    this.controller.setTextAreaTabindex(value);
-  }
-
-  render() {
-    const {
-      textAreaId,
-      ariaLabel,
-      ariaLabelledby,
-      cols,
-      disabled,
-      hasFocus,
-      textInputTabIndex,
-      name,
-      placeholder,
-      readOnly,
-      required,
-      rows,
-      spellcheck,
-      value,
-    } = this;
-
-    const labelId = ariaLabelledby || `${textAreaId}-label`;
-
+  render(): FunctionalComponent {
     return (
-      <Host {...{
-        class: {
-          'ods-error': Boolean(this.controller.hasError()),
-        },
-        tabindex: textInputTabIndex,
-        hasFocus,
-      }}
-      >
+      <Host class={{ 'ods-error': this.internalError }}>
```

Comment:
I think we need the tabindex here or the component will not be reflect on the host